src/replication.c

```diff
@@ -2386,6 +2481,353 @@ char *sendCommandArgv(connection *conn, int argc, char **argv, size_t *argv_lens
     return NULL;
 }
 
+sds getReplicaPortString(void) {
+    long long replica_port;
+    if (server.slave_announce_port) {
+        replica_port = server.slave_announce_port;
+    } else if (server.tls_replication && server.tls_port) {
+        replica_port = server.tls_port;
+    } else {
+        replica_port = server.port;
+    }
+    return sdsfromlonglong(replica_port);
+}
+
+void freeReplDataBuf(void) {
```

Comment:
function name is too generic.
```suggestion
void freePendingReplDataBuf(void) {
```